### ISSUE TITLE
Loss depends on batched vs online

### ISSUE THREAD

**Issue by ibmua** [BVLC/caffe#4409]
I have a network trained on data that's "not too" shuffled. So my network seems to have learned that stuff that was positioned near during training looks similar to the actual stuff, although it usually doesn't. When I train with batches (any >1 sample batch) I get loss <1, while when I do per-image training ("online training") I get loss usually >10. I found it a very interesting phenomenon, does anybody know how it happens that the loss gets so wrong?

**Comment by seanbell** [comment]
You should read about SGD and the effect of batch size.  Also, the loss value depends on whether or not your loss layer is averaging across the batch size.  Usually you want larger batch sizes and SGD won't learn a good solution unless either the batch size is medium-sized (~128 images) or your learning rate is very small.  Learning rate and batch size need to be changed together because smaller batch sizes are less stable (and thus need a smaller learning rate).

That being said, the issues page isn't the place for this discussion.  Please discuss modeling issues on the mailing list, thanks!

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
